cpp/src/link_analysis/pagerank.cu

```diff
@@ -350,23 +351,29 @@ void pagerank(raft::handle_t const &handle,
               bool has_guess)
 {
   CUGRAPH_EXPECTS(pagerank != nullptr, "Invalid API parameter: Pagerank array should be of size V");
-
+  // Multi-GPU
   if (handle.comms_initialized()) {
-    std::cout << "\nINSIDE CPP\n";
-    auto &comm = handle.get_comms();
-    std::cout << comm.get_rank() << "\n";
-    std::cout << comm.get_size() << "\n";
-    return;
-  }
-  return detail::pagerank_impl<VT, ET, WT>(graph,
-                                           pagerank,
-                                           personalization_subset_size,
-                                           personalization_subset,
-                                           personalization_values,
-                                           alpha,
-                                           tolerance,
-                                           max_iter,
-                                           has_guess);
+    CUGRAPH_EXPECTS(personalization_subset == nullptr,
+                    "Invalid API parameter: Multi-GPU Pagerank does not support Personalized "
+                    "variant, please use the single GPU version for this feature");
+    CUGRAPH_EXPECTS(
+      tolerance > 1e-6 && tolerance < 1e-4,
```

Comment:
+1, if tolerance is 0, it will never triggered.

Reply:
> This is an oddly specific way of checking tolerance.

Indeed, that's because the default value for a single GPU is 1e-5. This the value we are trying to identify here. There are a few other ways to do that but none is really concise.

> Specifying a tolerance of 0.01 or 0.00000001 will not trigger the error.
I think I am missing something. 0.01 > 1e-4 and  0.00000001 < 1e-6 they should both trigger the error.

> Would tolerance == WT{0} be sufficient?
I don't think so because it is not expected to be 0.

Reply:
So are we supporting tolerance but only allowing the value between [1e-6, 1e-4]? This sounds like an odd requirement? Why do we need this?

Reply:
The default value for a single GPU is 1e-5

Reply:
I get the default value, but we don't allow users to override default value?

Reply:
Users are allowed to override tolerance in SG path but not MG.
Any suggestions on how to better detect that tolerance was not overridden by the user in the MG path?
This is likely just for a few weeks as we will likely support tolerance soon.

Reply:
I'm confused.  The error message is indicates that OPG pagerank does not support tolerance.  This check, as written, will throw an exception if a tolerance of 1e-5 is passed in.

If you pass in a tolerance of 0.01 then it will check:
* 0.01 > 1e-6: TRUE
* 0.01 < 1e-4: FALSE
* TRUE && FALSE == FALSE  ... thus the exception will not be thrown.  Right?

Reply:
> Users are allowed to override tolerance in SG path but not MG.

Why? To avoid reduction? We should support this for both SG and MG.

Reply:
> > Users are allowed to override tolerance in SG path but not MG.

> Why? To avoid reduction? We should support this for both SG and MG.

https://github.com/rapidsai/cugraph/pull/947#discussion_r442914612

Reply:
So let me add some context : 
- Same API entry point for SG and OPG
- SG supports tolerance but OPG doesn't currently (but should soon)
- SG uses 1e-5 as default tolerance.

Temporarily, the idea was to throw an error in the MG path when tolerance was not 1e-5 (ie. when we can assume it was set by the user) but I'm open to other suggestions.

Notice that this only impacts C++ API users and will go away before a few weeks.

Reply:
If it's going away that soon, why even have the check?  But if it's going away soon, I'm fine either way.

Reply:
Ok, removed